### DISCUSSION TITLE
Language server plans?

### DISCUSSION THREAD

**Discussion by kahnclusions** [uwu-tech/Kind#300] [Ideas]
Are there any plans to build a language server implementing LSP? This would be a great way to provide support in various editors like VS Code, (Neo)Vim, etc.

Also, I don't know how this is the first time I've heard of the Kind project, but it looks like an amazing piece of work!

**Comment by racs4** [comment]
A project for this was being done at https://github.com/simonhorlick/kind-vscode, but I think it's stalled.
Recently I was trying to update it and I even managed to get a type check to work when changing a file, but I ended up not working on it anymore due to lack of time.
The repository uses this kind fork https://github.com/simonhorlick/Kind (which is also out of date), that has as its main change, a folder called LSP, where are the codes that will be compiled for JS and called in the LSP located in https://github.com/simonhorlick/kind-vscode

**Reply by rigille** [reply]
do you have your partial progress published somewhere @racs4?

**Reply by racs4** [reply]
No, but i plan to make a PR someday

**Comment by racs4** [comment]
I made the PR updating the extension to the newest kind version. Here it is: https://github.com/simonhorlick/kind-vscode/pull/24, I appreciate any suggestion
